pulp_2_tests/tests/docker/api_v2/test_remove.py

```diff
@@ -0,0 +1,154 @@
+# coding=utf-8
+"""Tests for removing docker units."""
+import unittest
+from urllib.parse import urljoin
+
+from packaging.version import Version
+from pulp_smash import api, config
+from pulp_smash.pulp2.constants import REPOSITORY_PATH
+from pulp_smash.pulp2.utils import search_units, sync_repo
+
+from pulp_2_tests.constants import DOCKER_V2_FEED_URL
+from pulp_2_tests.tests.docker.api_v2.utils import gen_repo
+from pulp_2_tests.tests.docker.utils import get_upstream_name
+
+# Dummy Data - Need to datamine a DOCKER repo
+DOCKER_REMOVE = {
+    'INITIAL': {'MANIFEST': 10, 'MANIFEST_LIST': 2, 'BLOB': 8, 'TAG': 13},
+}
+
+
+class RemoveV2ContentTestCase(unittest.TestCase):
+    """Ensure content removal of Docker repository information.
+
+    With the refactor of the docker importer's remove function to
+    increase performance, content removal needs to be functional verified.
+
+    The cases covered with content post-count verification for all units:
+
+    1. Remove all tags.
+    2. Remove all manifest_lists.
+    3. Remove all manifests.
+    4. Remove some non-shared manifest lists.
+    5. Remove some non-shared manifest.
+    6. Remove some shared manifests lists and verify shared units are not
+       recursively removed.
+    7. Remove some shared manifests and verify shared units are not
+       recursively removed.
+    8. Sync Repo A with fixture-1. Sync Repo A with fixture-2 with
+       Mirror=True. Verify Repo A's content counts exactly match fixture-2 and
+       not fixture-1.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Set variables used by each test case."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        if cls.cfg.pulp_version < Version('2.21'):
+            raise unittest.SkipTest(
+                'This test requires Pulp 2.21 or newer.'
+            )
+
+    def setUp(self):
+        """Set variables used by each test case."""
+        self.body = gen_repo()
```

Comment:
Not a blocker, but we can do:

```py
self.body = gen_repo(
    importer_config={
        'enable_v1': False,
        'enable_v2': True,
        'feed': DOCKER_V2_FEED_URL,
        'upstream_name': get_upstream_name(self.cfg),
    }
)
```

Reply:
@rochacbruno Ha! Yeah, I totally missed that. Thanks! Will get that in shortly + retested.